dace/sdfg/sdfg.py

```diff
@@ -1702,6 +1744,136 @@ def add_datadesc(self, name: str, datadesc: dt.Data, find_new_name=False) -> str
 
         return name
 
+    def temp_pgrid_name(self):
+        """ Returns a temporary process-grid name that can be used in this SDFG. """
+
+        name = '__pgrid%d' % self._pgrids_count
+        while name in self._pgrids:
+            self._pgrids_count += 1
+            name = '__pgrid%d' % self._pgrids_count
+        self._pgrids_count += 1
+
+        return name
+
+    def add_pgrid(self,
+                  shape: ShapeType = None,
+                  parent_grid: str = None,
+                  color: Sequence[Union[Integral, bool]] = None,
+                  exact_grid: RankType = None,
+                  root: RankType = 0):
+        """ Adds a process-grid to the process-grid descriptor store.
+            For more details on process-grids, please read the documentation of the ProcessGrid class.
+            :param shape: Shape of the process-grid (see `dims` parameter of [MPI_Cart_create](https://www.mpich.org/static/docs/latest/www3/MPI_Cart_create.html)), e.g., [2, 3, 3].
+            :param parent_grid: Parent process-grid (similar to the `comm` parameter of [MPI_Cart_sub](https://www.mpich.org/static/docs/v3.2/www3/MPI_Cart_sub.html)).
+            :param color: The i-th entry specifies whether the i-th dimension is kept in the sub-grid or is dropped (see `remain_dims` input of [MPI_Cart_sub](https://www.mpich.org/static/docs/v3.2/www3/MPI_Cart_sub.html)).
+            :param exact_grid: If set then, out of all the sub-grids created, only the one that contains the rank with id `exact_grid` will be utilized for collective communication.
+            :param root: Root rank (used for collective communication).
+            :return: Name of the new process-grid descriptor.
+        """
+
+        if not (shape or parent_grid):
+            raise ValueError("Process-grid must either have its shape defined or be linked to a parent-grid.")
+
+        # convert strings to int if possible
+        shape = shape or []
+        newshape = []
+        for s in shape:
+            try:
+                newshape.append(int(s))
+            except:
+                newshape.append(dace.symbolic.pystr_to_symbolic(s))
+        shape = newshape
+
+        grid_name = self.temp_pgrid_name()
+        is_subgrid = (parent_grid is not None)
+        if parent_grid and isinstance(parent_grid, str):
+            parent_grid = self._pgrids[parent_grid]
+
+        self._pgrids[grid_name] = ProcessGrid(grid_name, is_subgrid, shape, parent_grid, color, exact_grid, root)
+
+        self.append_init_code(self._pgrids[grid_name].init_code())
+        self.append_exit_code(self._pgrids[grid_name].exit_code())
+
+        return grid_name
+
+    def temp_subarray_name(self):
```

Comment:
all of these methods have a replacement already: `find_new_name` https://github.com/spcl/dace/blob/master/dace/data.py#L99

Reply:
I just noticed that some merge conflict led to that function appearing there twice. could you also remove the older one of them in this PR?

Reply:
`temp_data_name` is still there and used though

Reply:
If by duplicate you mean `dace::SDFG::_find_new_name` and `data::find_new_name`, then both are used.